src/app/features/location/address-gps-location/address-gps-location.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component } from "@angular/core";
+import { Logging } from "app/core/logging/logging.service";
+import { GpsService } from "../gps.service";
+import { MatTooltip } from "@angular/material/tooltip";
+import { FaIconComponent } from "@fortawesome/angular-fontawesome";
+import { MatProgressSpinnerModule } from "@angular/material/progress-spinner";
+import { MatIconButton } from "@angular/material/button";
+import { NgIf } from "@angular/common";
+import { MapPopupComponent } from "../map-popup/map-popup.component";
+import { Coordinates } from "../coordinates";
+import { AlertService } from "app/core/alerts/alert.service";
+
+@Component({
+  selector: "app-address-gps-location",
+  standalone: true,
+  imports: [
+    MatTooltip,
+    FaIconComponent,
+    MatProgressSpinnerModule,
+    NgIf,
+    MatTooltip,
+    MatIconButton,
+  ],
+  templateUrl: "./address-gps-location.component.html",
+  styleUrl: "./address-gps-location.component.scss",
+})
+export class AddressGpsLocationComponent {
+  constructor(
+    private gpsService: GpsService,
+    private mapPopupComponent: MapPopupComponent,
+    private alertService: AlertService,
+  ) {}
+  location: Coordinates;
+  public gpsLoading = false;
+
+  async updateLocationFromGps() {
+    this.gpsLoading = true;
+    try {
+      const location = await this.gpsService.getGpsLocationCoordinates();
+      if (location) {
+        this.location = {
+          lat: location.latitude,
+          lon: location.longitude,
+        };
+        await this.mapPopupComponent.mapClicked(this.location);
```

Comment:
In order to reuse this in a clean way, we shouldn't inject the component but instead move the method into a service so that it is properly encapsulated (Single Responsibility Principle --> a component shouldn't provide general methods in addition to defining UI).

I suggest you use GeoService.reverseLookup here and change that reverseLookup method, so that it includes the fallback logic that currently is done in MapPopupComponent.lookupCoordinates (https://github.com/Aam-Digital/ndb-core/blob/85e527475199e26d0b17c868d0acf1acec1e26b9/src/app/features/location/map-popup/map-popup.component.ts#L111-L118).
Ideally mapClicked there can then directly use geoService.reverseLookup. And updateLocationFromGps can also use that same method without extra transformations.

Reply:
I have updated the `reverseLookup` method as per your suggestion and added fallback logic, which was previously happening in `MapPopupComponent.lookupCoordinates`. I have one question here, do I also need to add the `updateLocation` logic in this `address.gps.location` component to help update the selected location field with the address based on coordinates?

Reply:
changes look good, thanks!

Reply:
As the GPS button is now a component, it should have an `@Output` property to emit whenever a new location was selected through that. Then you'll need to use that output in adress-edit.component to set the value. Maybe you can use the updateFromAddressSearch method. Please test that and see if it would need any adjustments

Reply:
Thanks, @sleidig for the suggestion. I will make the changes accordingly.

Reply:
nice, looks good!